examples/cute/tutorial/tiled_copy_sycl.cpp

```diff
@@ -0,0 +1,248 @@
+  // Given a 2D shape, perform an efficient copy
+  //
+  using dtype = float;
+
+  // Define a tensor shape with dynamic extents (m, n)
+  auto tensor_shape = make_shape(256, 512);
+
+  //
+  // Allocate and initialize
+  //
+  std::vector<dtype> host_src(size(tensor_shape));
+  std::vector<dtype> host_output(size(tensor_shape));
+
+  dtype* device_src = (dtype*)syclcompat::malloc(size(tensor_shape) * sizeof(dtype));
+  dtype* device_output = (dtype*)syclcompat::malloc(size(tensor_shape) * sizeof(dtype));
+
+  for (size_t i = 0; i < host_src.size(); ++i) {
+    host_src[i] = static_cast<dtype>(i);
+  }
+
+  syclcompat::memcpy(device_src, host_src.data(), size(tensor_shape) * sizeof(dtype));
+  syclcompat::memcpy(device_output, host_output.data(), size(tensor_shape) * sizeof(dtype));
+
+  //
+  // Make tensors
+  //
+
+  Tensor tensor_S = make_tensor(make_gmem_ptr(device_src), make_layout(tensor_shape));
+  Tensor tensor_D = make_tensor(make_gmem_ptr(device_output), make_layout(tensor_shape));
+
+  //
+  // Tile tensors
+  //
+
+  // Define a statically sized block (M, N).
+  // Note, by convention, capital letters are used to represent static modes.
+  auto block_shape = make_shape(Int<128>{}, Int<64>{});
+
+  if ((size<0>(tensor_shape) % size<0>(block_shape)) ||
+      (size<1>(tensor_shape) % size<1>(block_shape))) {
+    std::cerr << "The tensor shape must be divisible by the block shape." << std::endl;
+    return -1;
+  }
+  // Equivalent check to the above
+  if (not weakly_compatible(block_shape, tensor_shape)) {
+    std::cerr << "Expected the tensors to be weakly compatible with the block_shape." << std::endl;
+    return -1;
+  }
+
+  // Tile the tensor (m, n) ==> ((M, N), m', n') where (M, N) is the static tile
+  // shape, and modes (m', n') correspond to the number of tiles.
+  //
+  // These will be used to determine the CUDA kernel grid dimensions.
+  Tensor tiled_tensor_S = tiled_divide(tensor_S, block_shape);  // ((M, N), m', n')
+  Tensor tiled_tensor_D = tiled_divide(tensor_D, block_shape);  // ((M, N), m', n')
+
+  // Thread arrangement
+  Layout thr_layout = make_layout(make_shape(Int<32>{}, Int<8>{}));
+
+  // Vector dimensions
+  Layout vec_layout = make_layout(make_shape(Int<4>{}, Int<1>{}));
+
+  //
+  // Determine grid and block dimensions
+  //
+
+  auto gridDim =
+      syclcompat::dim3(size<1>(tiled_tensor_D),
+                       size<2>(tiled_tensor_D));  // Grid shape corresponds to modes m' and n'
+  auto blockDim = syclcompat::dim3(size(thr_layout));
+
+  //
+  // Launch the kernel
+  //
+  syclcompat::launch<copy_kernel_vectorized<decltype(tiled_tensor_S), decltype(tiled_tensor_D),
+                                            decltype(thr_layout), decltype(vec_layout)>>(
+      gridDim, blockDim, tiled_tensor_S, tiled_tensor_D, thr_layout, vec_layout);
+  syclcompat::wait_and_throw();
+
+  //
+  // Verify
+  //
+
+  syclcompat::memcpy(host_output.data(), device_output, size(tensor_shape) * sizeof(dtype));
```

Comment:
```suggestion
  syclcompat::memcpy<dtype>(host_output.data(), device_output, size(tensor_shape));
```

Reply:
Same reasoning as the one with `malloc`, it's more closer to cuda (`cudaMemcpy`)